tracer/src/Datadog.Trace/AppSec/SecuritySettings.cs

```diff
@@ -46,6 +49,12 @@ public SecuritySettings(IConfigurationSource source)
 
                 WafTimeoutMicroSeconds = (ulong)wafTimeout;
             }
+
+            var obfuscationParameterKeyRegex = source?.GetString(ConfigurationKeys.AppSecObfuscationParameterKeyRegex);
+            ObfuscationParameterKeyRegex = string.IsNullOrWhiteSpace(obfuscationParameterKeyRegex) ? ObfuscationParameterKeyRegexDefault : obfuscationParameterKeyRegex;
+
+            var obfuscationParameterValueRegex = source?.GetString(ConfigurationKeys.AppSecObfuscationParameterValueRegex);
+            ObfuscationParameterValueRegex = string.IsNullOrWhiteSpace(obfuscationParameterValueRegex) ? ObfuscationParameterValueRegexDefault : obfuscationParameterValueRegex;
```

Comment:
Is there any need to test that the user-supplied regex is a _valid_ regex? 🤔 How does the WAF handle it if you give it something like `@"(password"` for example?

Reply:
The WAF has a hard coded fall regex, that activates if the user passes an invalid regex. The WAF also outputs a log saying the regex is invalid.

It uses the RE2 syntax, so I don't believe we have a good way to validate the regex in the library.